src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeKeepAliveTests.cs

```diff
@@ -0,0 +1,198 @@
+
+        [TestMethod]
+        public void Dispose_DisposesTimer()
+        {
+            var timer = new Mock<ITimer>();
+
+            var testSubject = CreateTestSubject(timer: timer.Object);
+            testSubject.Dispose();
+
+            timer.Verify(x => x.Dispose(), Times.Once);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void EventRaised_KeepAliveSentOnlyIfProcessIsRunning(bool isProcessRunning)
+        {
+            var context = KeepAliveWorkflowContext.Create(isProcessRunning);
+            context.ResetCalledMethod();
+
+            context.RaiseTimerEvent();
+
+            // Only expecting the keep-alive if the process is running
+            if (isProcessRunning)
+            {
+                context.CheckOnlyExpectedMethods("Stop", "GetAsync", "Start");
+
+                var uriArg = context.HttpWrapper.Invocations.First().Arguments.First();
+                uriArg.ToString().EndsWith("/status").Should().BeTrue();
+            }
+            else
+            {
+                context.CheckOnlyExpectedMethods("Stop", "Start");
+            }
+        }
+
+        [TestMethod]
+        public void EventHandler_NonCriticalException_IsSuppressed()
+        {
+            var context = KeepAliveWorkflowContext.Create(isProcessRunning: true);
+            context.ResetCalledMethod();
+
+            context.Process.Setup(x => x.Start())
+                .Throws(new InvalidOperationException("this is a test"));
+
+            // We have to call the internal handler method here: if we just raise
+            // the event, the Forget() method means event handler returns before
+            // the exception is thrown.
+            Func<Task> act = async () => await context.DirectlyInvokeTimerHandlerAsync();
+
+            act.Should().NotThrow();
+
+            // Should restart the timer
+            context.CheckOnlyExpectedMethods("Stop", "Start");        
+        }
+
+        [TestMethod]
+        public void EventHandler_CriticalException_IsNotSuppressed()
+        {
+            var exceptionThrowingMethodCalled = new ManualResetEvent(false);
+
+            var context = KeepAliveWorkflowContext.Create(isProcessRunning: true);
+            context.ResetCalledMethod();
+
+            context.Process.Setup(x => x.Start())
+                .Throws(new StackOverflowException("this is a test"));
+
+            // We have to call the internal handler method here: if we just raise
+            // the event, the Forget() method means event handler returns before
+            // the exception is thrown.
+            Func<Task> act = async () => await context.DirectlyInvokeTimerHandlerAsync();
+
+            act.Should().ThrowExactly<StackOverflowException>().And.Message.Should().Be("this is a test");
+
+            // Should restart the timer
+            context.CheckOnlyExpectedMethods("Stop", "Start");
+        }
+
+        private class KeepAliveWorkflowContext
```

Comment:
There are several tests that need to check the order of method invocation. This helper class hides the setup-ugliness so the tests themselves can be simpler.